numpy/core/tests/test_arrayprint.py

```diff
@@ -172,11 +172,23 @@ def test_structure_format(self):
         # for issue #5692
         A = np.zeros(shape=10, dtype=[("A", "M8[s]")])
         A[5:].fill(np.datetime64('NaT'))
-        assert_equal(np.array2string(A),
-                "[('1970-01-01T00:00:00',) ('1970-01-01T00:00:00',) " +
-                "('1970-01-01T00:00:00',)\n ('1970-01-01T00:00:00',) " +
-                "('1970-01-01T00:00:00',) ('NaT',) ('NaT',)\n " +
-                "('NaT',) ('NaT',) ('NaT',)]")
+        assert_equal(
+            np.array2string(A),
+            texetwrap.dedent("""\
+            [('1970-01-01T00:00:00',) ('1970-01-01T00:00:00',) ('1970-01-01T00:00:00',)
+             ('1970-01-01T00:00:00',) ('1970-01-01T00:00:00',) ('NaT',) ('NaT',)
+             ('NaT',) ('NaT',) ('NaT',)]""")
```

Comment:
@ahaldane: Another formatting change for 1.14 needed here perhaps? Odd how timedelta aligns `NaT`, but datetime does not

Reply:
It definitely looks like some kind of historical accident, since timedelta format is implemented in python, while the datetime format just falls back to the scalar repr which is coded in C.

I think we should try to make it align if we have time before 1.14.